### PR TITLE
Blaze: switch to using canBlaze to determine Blaze eligibility

### PR DESCRIPTION
**Pull request by momo-ozawa**
Part of #20915
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/609

## Description
- Adds a new readonly  `canBlaze` property for the Blog model that determines blaze eligiblity based on the following:
    - The `can_blaze` option returned from `/me/sites` is true 
    - The user is an admin for the site
- Removes the `isBlazeApproved` property from the Blog model

## How to test

<details><summary>isAdmin</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. ✅ The Blaze dashboard card is displayed
3. ✅ The Blaze menu item is displayed
4. Switch to a site where you're not an admin
5. ✅ The Blaze dashboard card is NOT displayed
6. ✅ The Blaze menu item is NOT displayed

</p>
</details> 

<details><summary>Site visibility</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. ✅ The Blaze dashboard card is displayed
3. ✅ The Blaze menu item is displayed
4. Go to Site Settings > Privacy, and set the blog to private
5. ✅ The Blaze dashboard card is NOT displayed
6. ✅ The Blaze menu item is NOT displayed
7. Go to Site Settings > Privacy, and set the blog to public
8. ✅ The Blaze dashboard card is displayed
9. ✅ The Blaze menu item is displayed

</p>
</details> 

<details><summary>Posts list screen</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. Create three types of posts: public, private, and password protected
9. Go to the posts lists screen and select the published tab
10. On the public post, tap the ellipsis button
11. ✅ A "Promote with Blaze" alert action is displayed
12. On the private post, tap the ellipsis button
13. ✅ A "Promote with Blaze" alert action is NOT displayed
14. On the password protected post, tap the ellipsis button
15. ✅ A "Promote with Blaze" alert action is NOT displayed
16. On the posts list screen, switch to the drafts tab
17. On any post, tap the ellipsis button
18. ✅ A "Promote with Blaze" alert action is NOT displayed
19. Switch to a site that's not eligible for Blaze
20. On any post, tap the ellipsis button
21. ✅ A "Promote with Blaze" alert action is NOT displayed

</p>
</details> 

<details><summary>Pages list screen</summary>
<p>

1. Switch to a site that's eligible for Blaze (See peeHDf-zm-p2 for more info on Blaze requirements)
2. Create three types of pages: public, private, and password protected
3. Go to the pages lists screen and select the published tab
4. On the public page, tap the ellipsis button
6. ✅ A "Promote with Blaze" alert action is displayed
10. On the private page, tap the ellipsis button
11. ✅ A "Promote with Blaze" alert action is NOT displayed
12. On the password protected page, tap the ellipsis button
13. ✅ A "Promote with Blaze" alert action is NOT displayed
14. On the pages list screen, switch to the drafts tab
15. On any page, tap the ellipsis button
16. ✅ A "Promote with Blaze" alert action is NOT displayed
17. Switch to a site that's not eligible for Blaze
18. On any page, tap the ellipsis button
19. ✅ A "Promote with Blaze" alert action is NOT displayed

</p>
</details> 

## Regression Notes
1. Potential unintended areas of impact
Blaze eligibilty

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Followed the above testing steps

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: MIGRATIONS.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 This file documents changes in the data model. Please explain any changes to the
 data model as well as any custom migrations.
 
+## WordPress 150
+
+@momozw 2023-06-20
+
+- `Blog`: removed `isBlazeApproved` attribute.
```
Comment (kean) [inline MIGRATIONS.md:10]: I wonder if would make sense to deprecate it instead of removing to avoid the database migration?

Comment (momo-ozawa) [reply MIGRATIONS.md:10]: I'd prefer to remove isBlazeApproved all together, unless there's a clear benefit to deprecating / avoiding a lightweight migration. Wdyt @kean @hassaanelgarem?

**Review by hassaanelgarem — APPROVED** [review]
Works as described! 🚀

File: MIGRATIONS.md (hunk 1 above)
Comment (hassaanelgarem) [reply MIGRATIONS.md:10]: I'd go with removing it as well. The benefit I see to avoiding the migration is avoiding the added model file, which will increase the app size. But IMHO, that's insignificant and not worth the tech debt.

Comment (kean) [reply MIGRATIONS.md:10]: OK, sounds good to me. The main benefit to me is that it's easier to reason about the data model if there are as fewer versions as possible. But it's a moot point with so many already there. 

Every migration is a risk. The deprecated fields could be removed later when other major changes are applied to the model.

Comment (momo-ozawa) [reply MIGRATIONS.md:10]: Technically for lightweight migration I don't think we even need to do model versioning. On WPiOS we've been creating a new version for every change, but DayOneiOS doesn't. (Ref: p1666793197893439-slack-C02HT04NTU3)

> if there are as fewer versions as possible

True. I've started a convo on Slack about dropping model versioning for changes compatible with lightweight migrations (Ref: p1687364482569129-slack-C04U3GCAVK2)

**Comment by kean** [comment]
Tested the upgrade to this version – works well. Works offline too. I also tested on a site not eligible for Blaze.

**Review by kean — APPROVED** [review]

**Pull request merged**